src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -800,18 +824,33 @@ public EntityResult tAssignedEntity2Practitioner(AssignedEntity cdaAssignedEntit
 		if (!cdaAssignedEntity.getRepresentedOrganizations().isEmpty()) {
 			if (cdaAssignedEntity.getRepresentedOrganizations().get(0) != null
 					&& !cdaAssignedEntity.getRepresentedOrganizations().get(0).isSetNullFlavor()) {
-				org.hl7.fhir.dstu3.model.Organization fhirOrganization = tOrganization2Organization(
-						cdaAssignedEntity.getRepresentedOrganizations().get(0));
 
-				fhirPractitionerRole.setOrganization(getReference(fhirOrganization));
-				fhirPractitionerRole.setPractitioner(getReference(fhirPractitioner));
+				IEntryResult orgResult = tOrganization2Organization(
+						cdaAssignedEntity.getRepresentedOrganizations().get(0), bundleInfo);
+
+				if (orgResult.hasResult()) {
+					info.setOrgIsNew(true);
+				}
+				result.updateFrom(orgResult);
+
+				if (orgResult.getFullBundle() != null) {
+					org.hl7.fhir.dstu3.model.Organization fhirOrganization = FHIRUtil
+							.findFirstResource(orgResult.getFullBundle(), org.hl7.fhir.dstu3.model.Organization.class);
```

Comment:
same question as before and I imagine everywhere... we are saying if an org exists go get it from the bundle, but are we just picking the first one?

Reply:
We've talked about this since this comment, but just for future reference, the full bundle includes duplicates and the newResourceBundle (returned by .getBundle) does not. Whenever a transformation function is called, a new EntryResult is created with an empty FullBundle and empty NewResourceBundle. If the entry result does not have a new result, i.e. the resource was already created and stored in the bundleInfo, then the pre-existing organization is pulled from the BundleInfo and stored in the FullBundle for the purpose of retrieving a reference.

Reply:
The FullBundle will not contain additional resources than those created from the CDA object that was input as a function parameter.